src/solc_0.7/Game/GameMinter.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.7.5;
+pragma experimental ABIEncoderV2;
+
+import "../common/BaseWithStorage/WithMetaTransaction.sol";
+import "./GameToken.sol";
+import "../common/Interfaces/IGameMinter.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+
+contract GameMinter is WithMetaTransaction, IGameMinter {
+    ///////////////////////////////  Libs //////////////////////////////
+    using SafeMath for uint256;
+
+    ///////////////////////////////  Data //////////////////////////////
+
+    GameToken internal immutable _gameToken;
+    uint256 internal immutable _gameMintingFee;
+    uint256 internal immutable _gameUpdateFee;
+    address internal immutable _feeBeneficiary;
+    IERC20 internal immutable _sand;
+
+    ///////////////////////////////  Functions /////////////////////////
+
+    constructor(
+        GameToken gameTokenContract,
+        address metaTransactionContract,
+        uint256 gameMintingFee,
+        uint256 gameUpdateFee,
+        address feeBeneficiary,
+        IERC20 sand
+    ) {
+        _gameToken = gameTokenContract;
+        _setMetaTransactionProcessor(metaTransactionContract, METATX_SANDBOX);
+        _gameMintingFee = gameMintingFee;
+        _gameUpdateFee = gameUpdateFee;
+        _feeBeneficiary = feeBeneficiary;
+        _sand = sand;
+    }
+
+    /// @notice Function to create a new GAME token
+    /// @param from The address of the one creating the game, included in the gameId
```

Comment:
I need to update this.

Reply:
Yes. Should be much cheaper to eventually just upgrade a contract to remove admin rights than to manually migrate all the existing data from the old contract to a newer "Adminless" contract.

Reply:
Created a new jira task for this topic.
https://sandboxgame.atlassian.net/browse/TSBBLOC-182